src/lib/swapper/swappers/ThorchainSwapper/utils/longTailHelpers.ts

```diff
@@ -173,17 +173,17 @@ export const getQuotedAmountOutByPool = async (
 
 export const selectBestRate = (
   quotedAmounts: Map<Address, bigint>,
-): [Address, bigint] | undefined => {
+): [Address | undefined, bigint | undefined] => {
   return Array.from(quotedAmounts.entries()).reduce(
     (
-      addressWithHighestAmount: [Address, bigint] | undefined,
+      addressWithHighestAmount: [Address | undefined, bigint | undefined],
       [poolAddress, amount]: [Address, bigint],
     ) => {
-      if (addressWithHighestAmount === undefined) return [poolAddress, amount]
-      return amount > addressWithHighestAmount?.[1]
+      if (addressWithHighestAmount[1] === undefined) return [poolAddress, amount]
```

Comment:
Now that we changed the types for the first guy of the tupled to be maybe undefined, do we want to also add a check for it?

Reply:
You mean the second guy? It doesn't seems so useful as under the hood if `poolAddress` is defined then `amount` should too